provider/packet/environ_network.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package packet
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/context"
+	"github.com/juju/names/v4"
+)
+
+var _ environs.Networking = (*environ)(nil)
+
+// Subnets returns basic information about subnets known by the provider for
+// the environment.
+func (e *environ) Subnets(ctx context.ProviderCallContext, inst instance.Id, subnetIDs []network.Id) ([]network.SubnetInfo, error) {
+	subnets := []network.SubnetInfo{}
+
+	if inst != instance.UnknownId {
+		device, _, err := e.packetClient.Devices.Get(string(inst), nil)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		for _, n := range device.Network {
+			subnetID, cidr, err := makeSubnetIDForNetwork("device-"+device.Hostname, n.Address, strconv.Itoa(n.CIDR))
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+
+			subnet := network.SubnetInfo{
+				ProviderId:        network.Id(subnetID),
+				ProviderNetworkId: network.Id(n.ID),
+				CIDR:              cidr,
+				VLANTag:           0,
+			}
+
+			subnets = append(subnets, subnet)
+		}
+	} else {
+		projectID := e.cloud.Credential.Attributes()["project-id"]
+		ips, _, err := e.packetClient.ProjectIPs.List(projectID, nil)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		for _, ip := range ips {
+			subnetID, cidr, err := makeSubnetIDForNetwork("project-"+projectID, ip.Address, strconv.Itoa(ip.CIDR))
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+
+			subnet := network.SubnetInfo{
+				ProviderId:        network.Id(subnetID),
+				ProviderNetworkId: network.Id(ip.ID), //TODO: figure out what the network ID should be???
```

Comment:
@achilleasa I'm not sure what value to put here

Reply:
This is a provider-specific ID for the network itself; it can be safely omitted if it is not something meaningful for packet's implementation.